File: modules/spring-core/testdata/java/inspection/fileResource/src/com/ResourceService.java
package com;

import jakarta.annotation.PostConstruct;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.context.ApplicationContext;
import org.springframework.core.io.ClassPathResource;
import org.springframework.core.io.FileUrlResource;
import org.springframework.core.io.Resource;
import org.springframework.core.io.ResourceLoader;
import org.springframework.core.io.AbstractFileResolvingResource;
import org.springframework.stereotype.Service;

import java.net.MalformedURLException;

@Service
public class ResourceService {
    @Autowired
    ResourceLoader resourceLoader;
    @Autowired
    ApplicationContext applicationContext;
    @PostConstruct
    public void init() throws MalformedURLException {
        ClassPathResource path = new ClassPathResource("classpath:application1.properties");
        ResourceService.class.getResource("application1.properties");
        ResourceService.class.getClassLoader().getResource("/application1.properties");
        AbstractFileResolvingResource pathFile = new FileUrlResource("classpath:application1.properties");
        Resource res = resourceLoader.getResource("classpath:application1.properties");
        applicationContext.getResource("classpath:application1.properties");
    }
}

@Service
public class ResourceServiceVariables {
    @Autowired
    ResourceLoader resourceLoader;
    @Autowired
    ApplicationContext applicationContext;
    @PostConstruct
    public void init() throws MalformedURLException {
        final String PATH = "application1.properties";
        final String PATH1 = PATH;
        final String PATH2 = "/" + PATH1;
        final String PATH3 = "classpath:application1.properties";

        ClassPathResource path = new ClassPathResource(PATH3);
        ResourceService.class.getResource(PATH1);
        ResourceService.class.getClassLoader().getResource(PATH2);
        AbstractFileResolvingResource pathFile = new FileUrlResource(PATH3);
        Resource res = resourceLoader.getResource(PATH3);
        applicationContext.getResource(PATH3);
    }
}

@Service
public class ResourceServiceFields {
    final String PATH = "application1.properties";
    final String PATH1 = PATH;
    final String PATH2 = "/" + PATH1;
    final String PATH3 = "classpath:application1.properties";

    @Autowired
    ResourceLoader resourceLoader;
    @Autowired
    ApplicationContext applicationContext;

    @PostConstruct
    public void init() throws MalformedURLException {
        ClassPathResource path = new ClassPathResource(PATH3);
        ResourceService.class.getResource(PATH1);
        ResourceService.class.getClassLoader().getResource(PATH2);
        AbstractFileResolvingResource pathFile = new FileUrlResource(PATH3);
        Resource res = resourceLoader.getResource(PATH3);
        applicationContext.getResource(PATH3);
    }
}

File: modules/spring-core/testdata/kotlin/inspection/fileResource/src/com/ResourceService.kt
package com

import jakarta.annotation.PostConstruct
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.context.ApplicationContext
import org.springframework.core.io.ClassPathResource
import org.springframework.core.io.FileUrlResource
import org.springframework.core.io.ResourceLoader
import org.springframework.stereotype.Service
import java.net.MalformedURLException

@Service
class ResourceService {
    @Autowired
    var resourceLoader: ResourceLoader? = null

    @Autowired
    var applicationContext: ApplicationContext? = null

    @PostConstruct
    @Throws(MalformedURLException::class)
    fun init() {
        ClassPathResource("classpath:application1.properties")
        ResourceService::class.java.getResource("application1.properties")
        ResourceService::class.java.classLoader.getResource("/application1.properties")
        FileUrlResource("classpath:application1.properties")
        resourceLoader!!.getResource("classpath:application1.properties")
        applicationContext!!.getResource("classpath:application1.properties")
    }
}

@Service
class ResourceServiceVariables {
    @Autowired
    var resourceLoader: ResourceLoader? = null

    @Autowired
    var applicationContext: ApplicationContext? = null

    @PostConstruct
    @Throws(MalformedURLException::class)
    fun init() {
        val PATH = "application1.properties"
        val PATH1 = PATH
        val PATH2 = "/$PATH1"
        val PATH3 = "classpath:application1.properties"

        ClassPathResource(PATH3)
        ResourceServiceVariables::class.java.getResource(PATH1)
        ResourceServiceVariables::class.java.getClassLoader().getResource(PATH2)
        FileUrlResource(PATH3)
        resourceLoader!!.getResource(PATH3)
        applicationContext!!.getResource(PATH3)
    }
}

@Service
class ResourceServiceFields {
    val PATH: String = "application1.properties"
    val PATH1: String = PATH
    val PATH2: String = "/$PATH1"
    val PATH3: String = "classpath:application1.properties"

    @Autowired
    var resourceLoader: ResourceLoader? = null

    @Autowired
    var applicationContext: ApplicationContext? = null

    @PostConstruct
    @Throws(MalformedURLException::class)
    fun init() {
        ClassPathResource(PATH3)
        ResourceServiceFields::class.java.getResource(PATH1)
        ResourceServiceFields::class.java.getClassLoader().getResource(PATH2)
        FileUrlResource(PATH3)
        resourceLoader!!.getResource(PATH3)
        applicationContext!!.getResource(PATH3)
    }
}